### PR TITLE
Add sliding_window to MultiHeadAttention and GroupedQueryAttention

### PR DESCRIPTION
**Pull request by MarcosAsh**
## Description

Adds a `sliding_window` parameter to `MultiHeadAttention` and `GroupedQueryAttention`. When set to a positive integer, each query position attends to keys within `sliding_window - 1` positions on either side, producing the local-window pattern used by Mistral, Llama-3 long-context, and Phi-3. When combined with `use_causal_mask=True`, it yields the canonical causal sliding-window attention.

```python
mha = layers.MultiHeadAttention(num_heads=8, key_dim=64, sliding_window=128)
out = mha(query, value, use_causal_mask=True)  # causal sliding window
```

The mask is a symmetric band of width `2 * sliding_window - 1`, ANDed into the existing attention mask alongside the causal mask. This composes naturally with `attention_mask`, key/value masks, and causal masks.

The default `sliding_window=None` preserves existing behavior (full attention).

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/layers/attention/grouped_query_attention.py [hunk 1]
```diff
@@ -38,6 +38,12 @@ class GroupedQueryAttention(Layer):
         num_key_value_heads: Number of key and value attention heads.
         dropout: Dropout probability.
         use_bias: Boolean, whether the dense layers use bias vectors/matrices.
+        sliding_window: Optional positive integer. If set, restricts each
+            query position to attend only to the previous
+            `sliding_window - 1` key positions plus itself, producing the
```
[the remark this answers is not in the document]
Comment (MarcosAsh) [reply keras/src/layers/attention/grouped_query_attention.py]: Fixed

File: keras/src/layers/attention/grouped_query_attention.py [hunk 2]
```diff
@@ -392,6 +411,22 @@ def _compute_causal_mask(self, query, value=None):
...
+        ANDed with a causal mask, this yields the canonical causal
+        sliding-window pattern used by Mistral, Llama-3 long-context, and
+        Phi-3.
+        """
+        q_seq_length = ops.shape(query)[1]
+        v_seq_length = q_seq_length if value is None else ops.shape(value)[1]
+        ones_mask = ops.ones((1, q_seq_length, v_seq_length), dtype="int32")
+        row_index = ops.cumsum(ones_mask, axis=-2)
+        col_index = ops.cumsum(ones_mask, axis=-1)
+        return ops.less(ops.abs(row_index - col_index), self.sliding_window)
```
[the remark this answers is not in the document]
Comment (MarcosAsh) [reply keras/src/layers/attention/grouped_query_attention.py]: fixed

File: keras/src/layers/attention/multi_head_attention.py [hunk 3]
```diff
@@ -710,6 +730,22 @@ def _compute_causal_mask(self, query, value=None):
...
+        ANDed with a causal mask, this yields the canonical causal
+        sliding-window pattern used by Mistral, Llama-3 long-context, and
+        Phi-3.
+        """
+        q_seq_length = ops.shape(query)[1]
+        v_seq_length = q_seq_length if value is None else ops.shape(value)[1]
+        ones_mask = ops.ones((1, q_seq_length, v_seq_length), dtype="int32")
+        row_index = ops.cumsum(ones_mask, axis=-2)
+        col_index = ops.cumsum(ones_mask, axis=-1)
+        return ops.less(ops.abs(row_index - col_index), self._sliding_window)
```
[the remark this answers is not in the document]
Comment (MarcosAsh) [reply keras/src/layers/attention/multi_head_attention.py]: fixed

File: keras/src/layers/attention/multi_head_attention.py [hunk 4]
```diff
@@ -52,6 +52,12 @@ class MultiHeadAttention(Layer):
             feature dim (the query input's last dimension).
         attention_axes: axes over which the attention is applied. `None` means
             attention over all axes, but batch, heads, and features.
+        sliding_window: Optional positive integer. If set, restricts each
+            query position to attend only to the previous
+            `sliding_window - 1` key positions plus itself, producing the
```
[the remark this answers is not in the document]
Comment (MarcosAsh) [reply keras/src/layers/attention/multi_head_attention.py]: done

File: keras/src/layers/attention/multi_head_attention.py [hunk 5]
```diff
@@ -710,6 +730,25 @@ def _compute_causal_mask(self, query, value=None):
...
+        """
+        q_seq_length = ops.shape(query)[1]
+        v_seq_length = q_seq_length if value is None else ops.shape(value)[1]
+        row_index = ops.reshape(
+            ops.arange(q_seq_length, dtype="int32"), (1, q_seq_length, 1)
+        )
+        col_index = ops.reshape(
+            ops.arange(v_seq_length, dtype="int32"), (1, 1, v_seq_length)
+        )
+        return ops.less(ops.abs(row_index - col_index), self._sliding_window)
```
[the remark this answers is not in the document]
Comment (MarcosAsh) [reply keras/src/layers/attention/multi_head_attention.py:750]: I agree on the duplication but I think its out of scope for this pr. I can do this in another pr if necessary.

**Review by hertschuh — APPROVED** [review]
Thanks for adding this feature!
